modules/gsp-cluster/service-operator.tf

```diff
@@ -52,8 +52,24 @@ data "aws_iam_policy_document" "service-operator" {
     ]
 
     resources = [
-      "*" # TODO: revisit this
+      "*",
+    ] # TODO: revisit this
+  }
+
+  statement {
+    actions = [
+      "iam:*",
```

Comment:
```suggestion
      "iam:AttachRolePolicy",
      "iam:CreatePolicy",
      "iam:CreateRole",
      "iam:DeletePolicy",
      "iam:DeleteRole",
      "iam:DetachRolePolicy"
      "iam:PutRolePolicy"
      "iam:TagRole"
      "iam:UntagRole"
```
I worry about allowing `"iam:*"` because there's a lot of potential for privilege escalation. Not all IAM actions support the `iam:PermissionsBoundary` condition.  So, for example, someone could `iam:CreateAccessKey` on an existing User, even if that user were not subject to a permissions boundary (assuming such a user existed).  I'd rather be explicit about what we're allowing so I don't have to think about all these possible edge cases.